pkg/remediator/watch/filteredwatcher.go

```diff
@@ -155,6 +159,14 @@ func (w *filteredWatcher) ClearManagementConflictsWithKind(gk schema.GroupKind)
 	w.conflictHandler.ClearConflictErrorsWithKind(gk)
 }
 
+func (w *filteredWatcher) SetLatestCommitForGVK(commit string) {
```

Comment:
did you mean to store these in a map by gvk?

Seems like we kinda have to, since we may have GVKs from multiple commits being watched simultaneously.

Reply:
Each `filteredWatcher` is scoped to a single GVK, so by storing the commit on the `filteredWatcher` we are already mapping GVK to commit.

And yes the reason I went this route is because you might have commit A -> commit B where a certain GVK that was watched on commit A is no longer being watched on commit B.

Reply:
Ah, i see. The method name is just confusing because it doesn't take a gvk. Could use a comment, i guess.

Reply:
Given that the Runnable interface doesn't mention GVK anywhere else, I suggest just dropping ForGVK from the method names on the interface and method them in the method comment for the filteredWatcher methods specifically.

Reply:
Done.